proxy/src/transparency/server.rs

```diff
@@ -98,9 +102,30 @@ where
         let h2 = self.h2.clone();
         let tcp = self.tcp.clone();
         let new_service = self.new_service.clone();
-        let fut = connection
-            .peek_future(sniff)
-            .map_err(|_| ())
+        let timeout = Timeout::new(self.peek_timeout, &self.executor)
+            .expect("tokio Timeout::new failed");
+        let peek = connection.peek_future(sniff);
+        let fut = peek.select2(timeout)
```

Comment:
TIOLI, but I think you could also use the [proxy's `Timeout` module](https://github.com/runconduit/conduit/blob/master/proxy/src/timeout.rs) here. Although, I suppose the current implementation would require a `Service` impl for `peek_future`, which might not be ideal.

Reply:
In fact, I did start with trying to use that. But, I needed to get the underlying thing back if the timeout triggered, which our `timeout::TimeoutFuture` doesn't allow. Instead of changing everything that used that, I just settled for a select here.

Reply:
That's fair. I'm thinking about revisiting the `timeout` module eventually to make some other changes (i.e. moving to `tokio_timer` so we can use the hashed timing wheel, moving to a single timer thread, etc) , so I can try and make it better suited for this use-case as well.